pywt/_extensions/c/wt.template.c

```diff
@@ -447,20 +448,131 @@ int CAT(TYPE, _swt_)(TYPE input[], pywt_index_t input_len,
  * Approximation at specified level
  * input - approximation coeffs from upper level or signal if level == 1
  */
-int CAT(TYPE, _swt_a)(TYPE input[], pywt_index_t input_len, DiscreteWavelet* wavelet,
-                      TYPE output[], pywt_index_t output_len, int level){
+int CAT(TYPE, _swt_a)(TYPE input[], pywt_index_t input_len,
+                      DiscreteWavelet* wavelet,
+                      TYPE output[], pywt_index_t output_len,
+                      unsigned int level){
     return CAT(TYPE, _swt_)(input, input_len, wavelet->CAT(dec_lo_, TYPE),
                             wavelet->dec_len, output, output_len, level);
 }
 
 /* Details at specified level
  * input - approximation coeffs from upper level or signal if level == 1
  */
-int CAT(TYPE, _swt_d)(TYPE input[], pywt_index_t input_len, DiscreteWavelet* wavelet,
-                      TYPE output[], pywt_index_t output_len, int level){
+int CAT(TYPE, _swt_d)(TYPE input[], pywt_index_t input_len,
+                      DiscreteWavelet* wavelet,
+                      TYPE output[], pywt_index_t output_len,
+                      unsigned int level){
     return CAT(TYPE, _swt_)(input, input_len, wavelet->CAT(dec_hi_, TYPE),
                             wavelet->dec_len, output, output_len, level);
 }
 
+int CAT(TYPE, _swt_axis)(const TYPE * const restrict input, const ArrayInfo input_info,
```

Comment:
Except for ``swt_buffer_length``, ``_swt_a`` and ``_swt_d``, this whole function seems to be identical to some other functions above. Maybe with some extra template variables it can be avoided to add all this code?